promql/engine.go

```diff
@@ -1435,8 +1435,9 @@ func (ev *evaluator) VectorBinop(op ItemType, lhs, rhs Vector, matching *VectorM
 		// The rhs is guaranteed to be the 'one' side. Having multiple samples
 		// with the same signature means that the matching is many-to-many.
 		if _, found := rightSigs[sig]; found {
+			matchedLabels := rs.Metric.MatchLabels(matching.On, matching.MatchingLabels...)
 			// Many-to-many matching not allowed.
-			ev.errorf("many-to-many matching not allowed: matching labels must be unique on one side")
+			ev.errorf("duplicates found in labels: %s, many-to-many matching not allowed: matching labels must be unique on one side", matchedLabels.String())
```

Comment:
I'm thinking about how to make this error message as clear as possible to inexperienced users. Currently when I do:

    node_cpu_seconds_total / on(instance, job) node_cpu_seconds_total

...I get:

    Error executing query: duplicates found in labels: {instance="localhost:9100", job="node"}, many-to-many matching not allowed: matching labels must be unique on one side

How about changing it to something like:

    Error executing query: found multiple series for the match group {instance="localhost:9100", job="node"} on the right hand-side of the operation [or left-hand side, determine this based on matching.Card]: [include the two series here]; many-to-many matching not allowed: matching labels must be unique on one side